src/JuMPArray.jl

```diff
@@ -97,6 +97,20 @@ else
     Base.size(A::JuMPArray) = size(A.data)
     Base.LinearIndices(A::JuMPArray) = error("JuMPArray does not support this operation.")
     Base.axes(A::JuMPArray) = A.axes
+    Base.CartesianIndices(a::JuMPArray) = CartesianIndices(a.data)
+    struct JuMPArrayKey{T<:Tuple}
+        I::T
+    end
+    Base.getindex(k::JuMPArrayKey, index::Int) = getindex(k.I, index)
+    function JuMPArrayKey(ci::CartesianIndex, axes_array)
```

Comment:
I don't follow this function very easily. Could you add a comment explaining what it does?